sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/internal/state/MetricStorageUtilsTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.state;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.internal.aggregator.Aggregator;
+import io.opentelemetry.sdk.metrics.internal.aggregator.AggregatorHandle;
+import io.opentelemetry.sdk.metrics.internal.descriptor.MetricDescriptor;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class MetricStorageUtilsTest {
+
+  static final Attributes a = Attributes.of(AttributeKey.stringKey("a"), "a");
+  static final Attributes b = Attributes.of(AttributeKey.stringKey("b"), "b");
+  static final Attributes c = Attributes.of(AttributeKey.stringKey("c"), "c");
+  static final Attributes d = Attributes.of(AttributeKey.stringKey("d"), "d");
+  static final Attributes e = Attributes.of(AttributeKey.stringKey("e"), "e");
+  Map<Attributes, String> result;
+  Map<Attributes, String> toMerge;
+
+  @BeforeEach
+  void setup() {
+    result = new HashMap<>();
+    result.put(a, "A");
+    result.put(b, "B");
+    result.put(d, null);
+    toMerge = new HashMap<>();
+    toMerge.put(b, "B'");
+    toMerge.put(c, "C");
+    toMerge.put(e, null);
+  }
+
+  @Test
+  void mergeInPlace() {
+    Aggregator<String> agg = new TestAggregator();
+    MetricStorageUtils.mergeInPlace(result, toMerge, agg);
+
+    assertThat(result.keySet()).containsExactlyInAnyOrder(a, b, c, d);
+    assertThat(result.get(a)).isEqualTo("A");
+    assertThat(result.get(b)).isEqualTo("BB'");
+    assertThat(result.get(c)).isEqualTo("C");
+    assertThat(result.get(d)).isNull();
+  }
+
+  @Test
+  void diffInPlace() {
+    Aggregator<String> agg = new TestAggregator();
+    MetricStorageUtils.diffInPlace(result, toMerge, agg);
```

Comment:
I'm a bit confused by this test.  IIUC - you're using the presence of a `null` accumulation to denote that one of the values is gone?

That's not really what this is meant to do.  Accumulations (for well-behaved accumulators) should never be returning `null`.  `Diff` and `Merge` should just be `-` and `+`, in fact just using a SumAggregator directly might make this test easier to read/understand.

Given @anuraaga's push for using Nullaway, I'd prefer we don't encode tests that use `null` return values in Aggregator, as I think we need to go clean that up.

Reply:
would testing for `result.containsKey(xxx)` being false be a better assertion?

Reply:
Whoops, it was not my intention to have `diff()` return null in that mock -- I think I just overlooked it. Thanks for catching that. Let me fix that up.

Reply:
Sorry that it was probably confusing with that diff returning null....please have another look with the current form that uses a mock and is more explicit.